app/controllers/assessments_controller.rb

```diff
@@ -121,8 +121,21 @@ def install_assessment
         next
       end
 
-      # each assessment must have an associated yaml file
-      unless File.exist?(File.join(dir_path, filename, "#{filename}.yml"))
+      # each assessment must have an associated yaml file,
+      # and it must have a name field that matches its filename
+      if File.exist?(File.join(dir_path, filename, "#{filename}.yml"))
+        props = YAML.safe_load(File.open(
+                                 File.join(dir_path, filename, "#{filename}.yml"), "r", &:read
+                               ))
+        unless props["general"] && (props["general"]["name"] == filename)
+          flash.now[:error] = flash.now[:error] ? "#{flash.now[:error]} <br>" : ""
+          flash.now[:error] += "An error occurred while trying to display an existing assessment" \
```

Comment:
nit: need space between "assessment" and "on". Perhaps use "from" instead of "on"?

Reply:
Also could we exclude `name` from the yaml file altogether? Since `name` is already set and this seems like unnecessary work for instructors

This could be a separate PR, though

Reply:
Let's save that for a future PR